ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -88,17 +88,15 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
    * @param pollTimeoutMs the poll timeout.
    * @return a list of {@link ResponseInfo} representing the responses received for any requests that were sent out
    * so far.
-   * @throws IOException if the {@link Selector} associated with this NetworkClient throws
```

Comment:
We are trying to ensure that we don't aggressively close the router for exception from which we can possibly recover from - not necessarily isolate a particular operation.
There is no evidence that the exception from `poll()` is fatal and there is no way of knowing that so shutting down aggressively is worse than busy looping in the fatal case (when no productive work could be done) in my opinion. 

If we look at this another way: if the exception is indeed fatal, then the `selector` should/would close itself. This is a reasonable assumption to make, and the `NetworkClient` should check that and throw. Will push a change for that.